Subject: astgenkey should generate a private key that is not world-readable
Author: Lionel Elie Mamane <lionel@mamane.lu>
Bug: http://issues.asterisk.org/view.php?id=12373
Last-Update: 2009-12-19

Upstream has not accepted this patch and chose intead to document this 
as a known minor issue.

Index: asterisk-1.8.7.2/contrib/scripts/astgenkey
===================================================================
--- asterisk-1.8.7.2.orig/contrib/scripts/astgenkey	2011-12-14 08:03:50.393266055 -0500
+++ asterisk-1.8.7.2/contrib/scripts/astgenkey	2011-12-14 08:04:08.703265309 -0500
@@ -47,7 +47,11 @@
 rm -f ${KEY}.key ${KEY}.pub
 
 echo "Generating SSL key '$KEY': "
+oldumask="`umask`"
+umask 0077
 openssl genrsa -out ${KEY}.key ${DES3} 1024
+[ "$(id -u)" = 0 ] && chown asterisk: ${KEY}.key
+umask $oldumask
 openssl rsa -in ${KEY}.key -pubout -out ${KEY}.pub
 
 if [ -f "${KEY}.key" ] && [ -f "${KEY}.pub" ]; then
